duration.go

```diff
@@ -49,21 +49,29 @@ var (
 // Parse attempts to parse the given duration string into a *Duration,
 // if parsing fails an error is returned instead
 func Parse(d string) (*Duration, error) {
-	if !strings.Contains(d, "P") {
-		return nil, ErrUnexpectedInput
-	}
-
 	state := parsingPeriod
 	duration := &Duration{}
 	num := ""
 	var err error
 
-	for _, char := range d {
+	switch {
+	case strings.HasPrefix(d, "P"): // standard duration
+	case strings.HasPrefix(d, "-P"): // negative duration
+		duration.Negative = true
+	default:
+		return nil, ErrUnexpectedInput
+	}
+
+	for i, char := range d {
 		switch char {
 		case '-':
-			duration.Negative = true
+			if i != 0 {
+				return nil, ErrUnexpectedInput
```

Comment:
Why not simply call 

```go
d = strings.TrimLeft(d, "-")
```

after you do the HasPrefix(d, "-P")

Reply:
And so remove this case

Reply:
Nice suggestion